src/targets/gpu/prefuse_ops.cpp

```diff
@@ -124,34 +127,24 @@ struct find_add_layernorm
     }
 };
 
-#ifdef MIGRAPHX_USE_COMPOSABLEKERNEL
-
 struct pre_gemm_softmax_gemm : gemm_softmax_gemm
 {
     std::string name() const { return "gpu::pre_gemm_softmax_gemm"; }
 };
 MIGRAPHX_REGISTER_OP(pre_gemm_softmax_gemm);
 
-MIGRAPHX_PRED_MATCHER(is_ck_gemm, instruction_ref ins)
-{
-    if(ins->name() != "dot")
-        return false;
-    if(not pre_gemm_softmax_gemm::is_ck_supported_type(ins->get_shape().type()))
-        return false;
-    return true;
-}
-
+template <auto CheckDot>
 struct find_gemm_softmax_gemm
 {
     auto matcher() const
     {
         auto gemm1 =
-            match::skip(match::name("contiguous"))(match::name("dot")(is_ck_gemm().bind("gemm1")));
+            match::skip(match::name("contiguous"))(match::name("dot")(CheckDot().bind("gemm1")));
```

Comment:
This should just do `match::any_of(is_ck_gemm(), is_mlir_gemm())`.

Reply:
huh... how so ? 
is_ck_gemm is different is_mlir_gemm and ck would not want to offload what mlir wants to offload (vice versa).

Reply:
nvm, this works because you asked me later to move the enablements checks inside gemm checks.